azurerm/internal/services/cosmos/resource_arm_cosmosdb_account.go

```diff
@@ -1088,8 +1132,12 @@ func resourceAzureRMCosmosDBAccountGeoLocationHash(v interface{}) int {
 		}
 		location := azure.NormalizeLocation(m["location"].(string))
 		priority := int32(m["failover_priority"].(int))
+		zoneRedundant := false
+		if v, ok := m["zone_redundant"].(bool); ok {
+			zoneRedundant = v
+		}
 
-		buf.WriteString(fmt.Sprintf("%s-%s-%d", prefix, location, priority))
+		buf.WriteString(fmt.Sprintf("%s-%s-%d-%t", prefix, location, priority, zoneRedundant))
```

Comment:
This is a breaking change that requires a state migration to update the hashes for existing resources. See `ResourceAzureRMKeyVaultMigrateState` for an example of a state migration.

Reply:
I will work on a migration